src/txn/txn.c

```diff
@@ -925,9 +929,11 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
           true);
 
         if (ret == 0)
-            WT_ERR(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key, cbt->ref->page,
-              upd, commit, &fix_upd, &upd_appended));
-        else if (ret == WT_NOTFOUND && !commit) {
+            /* Not found if we cross the tree or key boundary. */
+            WT_ERR_NOTFOUND_OK(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key,
+                                 cbt->ref->page, upd, commit, &fix_upd, &upd_appended),
+              true);
+        if (ret == WT_NOTFOUND && !commit) {
```

Comment:
This was an else if which we previously couldn't get into but now I think we can, thats fine?

Reply:
The intention is to get into that after we call __txn_append_hs_record

Reply:
We return not found in __txn_append_hs_record if the cursor is placed on the wrong key. In this case, we also need to delete the prepared update.

Reply:
Thanks makes sense, thanks.